### ISSUE TITLE
Retrived empty data: what have I done wrong?

### ISSUE THREAD

**Issue by fgdfgfthgr-fox** [janelia-cellmap/cellmap-segmentation-challenge#117]
```

import math
import torch
import matplotlib.pyplot as plt
from cellmap_segmentation_challenge.utils import get_dataloader

if __name__ == "__main__":
    array_info = {
        "shape": (128, 128, 128),
        "scale": (8, 8, 8),
    }
    train_loader, val_loader = get_dataloader('datasplit.csv', ["mito",], 1,
                                              input_array_info=array_info, target_array_info=array_info, device='cpu',
                                              iterations_per_epoch=32)
    train_loader.refresh()
    loader = iter(train_loader.loader)
    example_samples = []
    for i in range (16):
        example_samples.append(next(loader))
    num_data = len(example_samples)
    plt.figure(figsize=(16, 9))
    rows = math.floor(math.sqrt(num_data * 2))
    cols = math.ceil(num_data / rows)
    for k in range(0, num_data):
        pair = example_samples[k]
        image = pair['input'][:, :, 64:64 + 1, :, :].squeeze()
        label = pair['output'][:, :, 64:64 + 1, :, :].squeeze()
        label = torch.nan_to_num(label)

        # Plot Image
        plt.subplot(rows, 2 * cols, 2 * k + 1)
        plt.imshow(image.cpu().numpy(), cmap='gist_gray')
        plt.axis('off')

        # Plot Label
        plt.subplot(rows, 2 * cols, 2 * k + 2)
        plt.imshow(label.cpu().numpy(), cmap='gist_gray')
        plt.axis('off')

    plt.subplots_adjust(wspace=0.1, hspace=0.1)
    plt.show()
```
I am trying to see what an example batch would look like so I write this simple script to help me visualize the samples. However, the images I got looks like this:

![Image](https://github.com/user-attachments/assets/c9a1b177-4f51-4143-ad1a-93a25cb58c5e)

Most of them are empty and the label doesn't match with the image. It seems like I didn't do anything different than the data retrieval in the train.py file, but somehow I couldn't get the correct data.

**Comment by rhoadesScholar** [comment]
Try using the same plotting function used in train.py.

**Comment by fgdfgfthgr-fox** [comment]
Same issue here.
```
train_loader, val_loader = get_dataloader('datasplit.csv', ["mito",], 8,
                                              input_array_info=array_info, target_array_info=array_info, device='cpu',
                                              iterations_per_epoch=32)
train_loader.refresh()
loader = iter(train_loader.loader)
pair = next(loader)
inputs = pair['input']
targets = pair['output']
outputs = pair['output']
figs = get_fig_dict(inputs, targets, outputs, ["mito",])
figs['mito'].show()
```

![Image](https://github.com/user-attachments/assets/335a4c23-1693-44e7-9a3d-28686825f4fc)

**Comment by rhoadesScholar** [comment]
This is looking like the same issue as https://github.com/janelia-cellmap/cellmap-segmentation-challenge/issues/110

**Comment by fgdfgfthgr-fox** [comment]
> This is looking like the same issue as [#110](https://github.com/janelia-cellmap/cellmap-segmentation-challenge/issues/110)
> 
> Is this from data that was downloaded as a zip?

Not exactly the same issue as setting target class to ["mito"] works fine for the included training script. But yes, it's from the data downloaded as a zip.

**Comment by rhoadesScholar** [comment]
> > This is looking like the same issue as [#110](https://github.com/janelia-cellmap/cellmap-segmentation-challenge/issues/110)
> > 
> > Is this from data that was downloaded as a zip?
> 
> Not exactly the same issue as setting target class to ["mito"] works fine for the included training script. But yes, it's from the data downloaded as a zip.

Interesting 🤔 I think I may know the issue. Are you sure you get non-empty raw data for all batches when using just ["mito"] as the target class in the training script? I suspect you will get some, but not all empty raw data because you didn't download the 8nm data for mito only crops.

**Comment by yuriyzubov** [comment]
Hi @fgdfgfthgr-fox 
@rhoadesScholar is correct, the data for 8nm is missing. By default, you only download the fibsem data resolution level that  matches the scale of the ground truth crops.
 
For example, when I ran your script with 32nm resolution (`array_info = {
        "shape": (128, 128, 128),
        "scale": (32, 32, 32),
    }`), all input raw fibsem data in this case loaded correctly: 

<img width="1035" alt="Image" src="https://github.com/user-attachments/assets/75673f4c-abb5-4adf-841c-0b0ec677ab01" />

In order to download all resolutions levels for the FIBSEM data, you need to run `csc fetch-data --fetch-all-em-resolutions`, (with `--zip` if you want compressed single file). The size of the file will be significantly larger. Instead of 39gb zip file, you'll download ~1.1TB zip file.    

Hope this would help to resolve data loading issue.

**Comment by fgdfgfthgr-fox** [comment]
Ok... Guess I will get a new drive then.

**Comment by fgdfgfthgr-fox** [comment]
> Hi [@fgdfgfthgr-fox](https://github.com/fgdfgfthgr-fox) [@rhoadesScholar](https://github.com/rhoadesScholar) is correct, the data for 8nm is missing. By default, you only download the fibsem data resolution level that matches the scale of the ground truth crops.
> 
So, you mean different classes have their ground truth label at different resolutions? When I downloaded the data without` --fetch-all-em-resolutions`, the em crops that contains only mito or other low resolution classes are in low resolution as well, while em crops that contain higher resolution labels are in high resolution. Is that correct?

**Comment by fgdfgfthgr-fox** [comment]
> Exactly correct.

You guys already answered me in the email, but thanks!
